include/deal.II/lac/petsc_matrix_base.h

```diff
@@ -1089,36 +1092,21 @@ namespace PETScWrappers
     Tmmult(MatrixBase &C, const MatrixBase &B, const VectorBase &V) const;
 
   private:
-    /**
-     * An internal array of integer values that is used to store the column
-     * indices when adding/inserting local data into the (large) sparse
-     * matrix.
-     *
-     * This variable does not store any "state" of the matrix
-     * object. Rather, it is only used as a temporary buffer by some
-     * of the member functions of this class. As with all @p mutable
-     * member variables, the use of this variable is not thread-safe
-     * unless guarded by a mutex. However, since PETSc matrix
-     * operations are not thread-safe anyway, there is no need to
-     * attempt to make things thread-safe, and so there is no mutex
-     * associated with this variable.
-     */
```

Comment:
This is an interesting comment given that the failing test 

https://github.com/dealii/dealii/blob/master/tests/petsc/assemble_matrix_parallel_01.cc

actually uses both `add()` and threads. We got lucky because the following happened:

1. `AffineConstraints::distribute_local_to_global()` does not elide zeros, so we do not copy the values
2. For the same reason we used `reinterpret_cast` to avoid copying any indices 

so, in practice, this was fine but the threaded version with `elide_zero_values = true` was always wrong.